src/core/api/rtd_csv_transaction/create-sas-token-policy.xml

```diff
@@ -63,18 +63,27 @@
 
             See: https://stackoverflow.com/questions/38474362/get-a-file-sha256-hash-code-and-checksum
         -->
-        <set-variable name="senderCode" value="@(((IResponse)context.Variables["senderCodeResponse"]).Body.As<string>())"/>
 
-        <set-variable name="senderCodeHash" value="@{
+        <choose>
+            <when condition="@(((IResponse)context.Variables["senderCodeResponse"]).StatusCode == 200)" >
+                <set-variable name="senderCode" value="@(((IResponse)context.Variables["senderCodeResponse"]).Body.As<string>())"/>
+            </when>
+            <otherwise>
+                <set-variable name="senderCode" value="unkno"/>
+            </otherwise>
+        </choose>
+
+        <set-variable name="keyHash" value="@{
                 System.Security.Cryptography.SHA256 hasher = System.Security.Cryptography.SHA256.Create();
-                return BitConverter.ToString(hasher.ComputeHash(System.Text.Encoding.UTF8.GetBytes(context.Variables.GetValueOrDefault("senderCode", "")))).Replace("-", "").ToLowerInvariant();
+                    return BitConverter.ToString(hasher.ComputeHash(System.Text.Encoding.UTF8.GetBytes(context.Request.Headers.GetValueOrDefault("Ocp-Apim-Subscription-Key","")))).Replace("-", "").ToLowerInvariant();
             }"
         />
         
         <set-variable name="containerName" value="@{
-            return string.Format("{0}-{1}",
-                (string)context.Variables["containerPrefix"],
-                ((string)context.Variables["senderCodeHash"]).Substring(0, 44));
+            return string.Format("{0}-{1}{2}",
```

Comment:
maybe we could place a separator between second and third argument, to make it more readable.

Reply:
You are right but it would cause pattern matching (`[a-z0-9]{44}`) problems during event evaluation.

Reply:
ok got it. Since it is not needed to frequently query directly the containers I don't think it's a problem. Thank you.